packages/grafana-ui/src/types/panel.ts

```diff
@@ -1,14 +1,22 @@
 import { ComponentClass } from 'react';
 import { LoadingState, SeriesData } from './data';
 import { TimeRange } from './time';
-import { ScopedVars } from './datasource';
+import { ScopedVars, DataRequestInfo, DataQueryError } from './datasource';
 
 export type InterpolateFunction = (value: string, scopedVars?: ScopedVars, format?: string | Function) => string;
 
+export interface QueryResponseData {
+  state: LoadingState;
```

Comment:
Only this caught my attention. I feel like loading state does not belong to response, as response is a consequence of loading state. On the other hand it's nicer to pass single `data` property to panel, than passing data and loading separately.

Reply:
it is a bit funny.  Maybe `ResponseData` isn't the right name -- the goal is an object that holds info about the request and response; what we asked for and what it got.  We don't want `loading` next to this because we will eventually have multiple instances (for annotations)

The loading state is weird because you could could really figure it out from the rest of the properties.
 * NotStarted = (!data || !data.series)
 * Error = (!!error)
 * Done = (request && series && request.endTime)

However in #16563, it sends partial updates and that would require something different for setting/clearing errors.  In that case we could remove the `state`, but make error an array?  so empty error list is explicitly clearing the error

Reply:
I think we should strive to always wrap loading state calculation in a sth like `LoadingState` type. Having i.e. `request && series && request.endTime` is cryptic and requires someone using the lib to exactly know what and how particular properties behave.

Reply:
yes, there is something with the loading state that feels wrong

Reply:
The naming here feels wrong, we already have a DataQueryResponseData 
https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/types/datasource.ts#L173  (the type that datasources return). 

The old PanelData type name maybe is better, or someo other name?

Reply:
I renamed it to PanelData and think it is much better now